packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -517,5 +518,20 @@ function scopeStorageService(
 		async list(path) {
 			return service.list(`${scope}${path}`);
 		},
+		getSnapshotTree(): ISnapshotTree | undefined {
+			const snapshotTree = service.getSnapshotTree?.();
+			if (snapshotTree === undefined) {
+				return undefined;
+			}
+			let scopedTree = snapshotTree;
+			for (const element of pathElements) {
+				const tree = scopedTree.trees[element];
+				if (tree === undefined) {
```

Comment:
I don't understand the use case for scopeStorageService, or IChannelStorageService.getSnapshotTree, but this seems a bit odd.

it seems like this expects some particular prefix on the path, and if it doesn't find it, it just returns the last node it was at while walking that path?

Is there a reason why that such a partial path traversal should be returned instead of erring with an assert? Is there something that uses this which depends on this partial traversal bring returned? If so, the docs for scopeStorageService seem like they should be updated, and if not it seems like this should assert instead of break.

Reply:
It's a mistake that it returns a partial path. It should assert if doesn't find a snapshot tree.

That function is slightly odd though. It could be better named and have better comments. I will make a note and do that as a separate PR.